src/main/util/file_util.py

```diff
@@ -0,0 +1,147 @@
+    return get_original_file_name(hashed_file_name) + '.' + extension
+
+
+def remove_file(file: str):
+    if os.path.isfile(file):
+        os.remove(file)
+
+
+def get_content_from_file(file: str):
+    with open(file, 'r') as f:
+        return f.read().rstrip("\n")
+
+
+def create_file(content: str, extension: str, file_name: str):
+    with open(file_name + '.' + extension, 'w') as f:
+        f.write(content)
+
+
+def create_directory(directory: str):
+    if not os.path.exists(directory):
+        os.makedirs(directory)
+        
+        
+def remove_directory(directory: str):
+    if os.path.exists(directory):
+        shutil.rmtree(directory, ignore_errors=True)
+
+
+# To get all files or subdirs (depends on the last parameter) from root that match item_condition
+# Can be used to get all codetracker files, all data folders, etc.
+# Note that all subdirs or files already contain the full path for them
+def get_all_file_system_items(root: str, item_condition, item_type=FILE_SYSTEM_ITEM.FILE.value):
+    items = []
+    for fs_tuple in os.walk(root):
+        for item in fs_tuple[item_type]:
+            if item_condition(item):
+                items.append(os.path.join(fs_tuple[FILE_SYSTEM_ITEM.PATH.value], item))
+    return items
+
+
+def csv_file_condition(name: str):
+    return get_extension_from_file(name) == "csv"
+
+
+# to get all codetracker files
+def ct_file_condition(name: str):
+    return ACTIVITY_TRACKER_FILE_NAME not in name and csv_file_condition(name)
+
+
+# to get all subdirs that contain ct and ati data
+def data_subdirs_condition(name: str):
+    return DATA_FOLDER_WITH_ATI in name or DATA_FOLDER_WITHOUT_ATI in name
+
+
+# to get path to the result folder that is near to the original folder
+# and has the same name but with a suffix added at the end
+def get_result_folder(folder: str, result_name_suffix: str):
+    result_folder_name = get_file_name_from_path(folder) + "_" + result_name_suffix
+    return os.path.join(get_parent_folder(folder), result_folder_name)
+
+
+def check_folder_and_write_df_to_file(folder_to_write: str, file_to_write: str, df: pd.DataFrame):
+    if not os.path.exists(folder_to_write):
+        os.makedirs(folder_to_write)
+
+    # get error with this encoding=ENCODING on ati_225/153e12:
+    # "UnicodeEncodeError: 'latin-1' codec can't encode character '\u0435' in position 36: ordinal not in range(256)"
+    # So change it then to 'utf-8'
+    try:
+        df.to_csv(file_to_write, encoding=ENCODING, index=False)
+    except UnicodeEncodeError:
+        df.to_csv(file_to_write, encoding='utf8', index=False)
+
+
+# to write a dataframe to the result_folder remaining the same file structure as it was before
+# for example, for file path/folder1/folder2/ati_566/file.csv the dataframe will be 
+# written to result_folder/folder1/folder2/ati_566/file.csv
+def write_result(result_folder: str, path: str, file: str, df: pd.DataFrame):
+    path_from_result_folder_to_file = file[len(path):]
```

Comment:
You should check not only the length of the path but also a coincidence of the common part because we can have the same length for two paths, but it may be the different parts